### ISSUE TITLE
Mobile meta tags

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#683]
@peller, @billreed63

The Dojo 1.7 Mobile Specification [1] shows that developers should include the following META tags in their HTML page:

```
<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,minimum-scale=1,user-scalable=no"/>
<meta name="apple-mobile-web-app-capable" content="yes" />
```

Currently, Maqetta doesn't include these META tags in the HTML pages we create. Seems like we need to add these two META tags. Need to do a little research to see if we need different META tags for Android versus iPhone.

[1] https://docs.google.com/document/pub?id=1lzdeIsZBMyU18t-Z4L8z6fqGkgJPnP9Aa29RZflYV9E

**Comment by JonFerraiolo** [comment]
I moved this from M3 to M2 because I think we need to have the META tags before shipping. The reason is that I tried to actually view a Maqetta-created UI mockup on my iPhone. (Method I used: I created an HTML file, then downloaded a ZIP with Dojo inside, uploaded to a web server, and then brought up the page on my iPhone.) Without the META tags, the page comes up as a standard desktop UI. With the META tags, it looks and runs as it should. For awhile, I'll leave the files:
- http://ferraiolo.com/file9/file9.html (without META tags)
- http://ferraiolo.com/file9/file9a.html (with META tags)

It is just so much more compelling if the UIs created in Maqetta actually work correctly on a mobile device.

**Comment by jhpedemonte** [comment]
The [OAM spec](http://www.openajax.org/member/wiki/OpenAjax_Metadata_1.0_Specification_Widget_Metadata#require_element) allows for a type attribute of `markup` for **require elements**.  We could specify this the View, ScrollableView and SwapView widgets:

``` xml
<require type="markup">
   <![CDATA[
      <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,minimum-scale=1,user-scalable=no"/>
      <meta name="apple-mobile-web-app-capable" content="yes" />
   ]]>
</require>
```

The code that adds this would obviously check for duplicates, based on the `name` attr to the `<meta>` tag.

The other option is to add some more code to the View **Helper** object to do the same.

**Comment by JonFerraiolo** [comment]
@jhpedemonte I was hoping you would figure out the best approach in dealing with OAM issues and I wouldn't have to think. :-) But now that I'm thinking, here are some options:

1 Main container widgets (View, etc.) include REQUIRE TYPE="markup" as you show above
2 All dojox.mobile widgets include REQUIRE TYPE="markup". Although most of the time you need a View, in a percentage of cases you don't. This approach has the advantage that each widget is self-contained, and for most things (e.g., included JS and CSS) each OAM file provides everything that a widget needs at runtime.
3 Instead of putting REQUIRE TYPE=markup in the OAM for View/etc, have a helper function add the META tags.
4 Instead of putting REQUIRE TYPE=markup in the OAM for all widgets, have a helper function add the META tags.
5 We invent a notion of "library metadata" and the REQUIRE TYPE=markup goes into the library metadata and gets inherited by all of the widgets in that library
6 dojox.mobile adds the META tags automatically as an automatic service similar to the logic that pulls in the correct theme CSS files
7 Maqetta core stuffs in META tags whenever a mobile silhouette is active
8 Maqetta provides UI at document creation time to let the user choose whether to include the META tags
9 Just document our way around this issue, and force the user to type in META tags in source view

I'm in favor of Maqetta and/or Dojo doing automatic action so users can focus on UI design and everything just works. 

Note that META tags are largely iOS specific. According to someone I know with a gPhone, you don't need the META tags on Android phones. There might be other META tags (or similar nonsense) for other devices in the future. 

I'm inclined to keep this out of the OAM and instead have Maqetta core logic add the two META tags whenever the user is currently using a silhouette, and remove the META tags when not using a silhouette. Down the road, maybe we offer a preference to allow the user to control this behavior.

**Comment by jhpedemonte** [comment]
From what I found, the meta tags are used by most mobile browsers, not just Safari on iOS.

I hadn't thought about tying the meta tags to the use of silhouettes.  That might work for now, but I feel like there might be a better solution.

**Comment by jhpedemonte** [comment]
So, not sure we want to use `apple-mobile-web-app-capable`.  According to the [docs](http://developer.apple.com/library/safari/#documentation/AppleApplications/Reference/SafariHTMLRef/Articles/MetaTags.html), that meta tag is used to run a web app in "full screen mode" -- that is, run it similar to how native apps are run.  If this isn't set, then the app runs in the web browser.  I think we want the latter behavior.

If the user wants to run the app full-screen, they'll have to add that meta tag themselves.

**Comment by JonFerraiolo** [comment]
Thanks for being thorough! I actually didn't research the META tags. I blinded copied/pasted from the example near the top of https://docs.google.com/document/pub?id=1lzdeIsZBMyU18t-Z4L8z6fqGkgJPnP9Aa29RZflYV9E&pli=1#h.ayvrhdms4pdv, and what you say above makes sense.

**Comment by jhpedemonte** [comment]
For reference, here are some docs on the mobile meta tags: http://www.html5rocks.com/en/mobile/mobifying.html#toc-meta.

**Comment by jhpedemonte** [comment]
This is working for me now.  Jon, let me know if I missed something in the previous checkins.

Side note: I created a couple of generic functions (`_addHeadElement` and `_removeHeadElement`) that we can hopefully use more in Context.js (M3 and onwards).  They add/remove on the Model and DOM at the same time.  The add function also checks for duplicates.  Hopefully, we can generalize some more, to work on the whole document, not just `<head>`.

**Comment by JonFerraiolo** [comment]
Bless you for creating a general purpose function
